imap_processing/ialirt/process_ephemeris.py

```diff
@@ -0,0 +1,226 @@
+    longitude: float,
+    latitude: float,
+    altitude: float,
+    observation_time: Union[float, np.ndarray],
+    target: SpiceBody = SpiceBody.IMAP.name,
+) -> tuple:
+    """
+    Calculate azimuth and elevation.
+
+    Parameters
+    ----------
+    longitude : float
+        Longitude in decimal degrees. Positive east of prime meridian,
+        negative to west.
+    latitude : float
+        Latitude in decimal degrees. Positive north of equator, negative
+        to south.
+    altitude : float
+        Altitude in kilometers.
+    observation_time : float or np.ndarray
+        Time at which the state of the target relative to the observer
+        is to be computed. Expressed as ephemeris time, seconds past J2000 TDB.
+    target : str (Optional)
+        The target body. Default is "IMAP".
+
+    Returns
+    -------
+    azimuth : np.ndarray
+        Azimuth in degrees.
+    elevation : np.ndarray
+        Elevation in degrees.
+    """
+    observer_position_ecef = latitude_longitude_to_ecef(longitude, latitude, altitude)
+
+    if not isinstance(observation_time, np.ndarray):
+        azel_results = spice.azlcpo(
+            method="Ellipsoid",  # Only method supported
+            target=target,  # target ephemeris object
+            et=observation_time,  # time of observation
+            abcorr="LT+S",  # Aberration correction
+            azccw=False,  # Azimuth measured clockwise from the positive y-axis
+            elplsz=True,  # Elevation increases from the XY plane toward +Z
+            obspos=observer_position_ecef,
+            # observer position relative to center of motion
+            obsctr="EARTH",  # Name of the center of motion
+            obsref="IAU_EARTH",
+            # Body-fixed, body-centered reference frame wrt observer's
+            # center
+        )
+        return np.rad2deg(azel_results[0][1]), np.rad2deg(azel_results[0][2])
+    else:
+        azimuth = []
+        elevation = []
+
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.azlcpo
+        for timestamp in observation_time:
+            azel_results = spice.azlcpo(
+                method="Ellipsoid",  # Only method supported
+                target=target,  # target ephemeris object
+                et=timestamp,  # time of observation
+                abcorr="LT+S",  # Aberration correction
+                azccw=False,  # Azimuth measured clockwise from the positive y-axis
+                elplsz=True,  # Elevation increases from the XY plane toward +Z
+                obspos=observer_position_ecef,  # observer pos. to center of motion
+                obsctr="EARTH",  # Name of the center of motion
+                obsref="IAU_EARTH",  # Body-fixed, body-centered reference frame wrt
+                # observer's center
+            )
+            azimuth.append(np.rad2deg(azel_results[0][1]))
+            elevation.append(np.rad2deg(azel_results[0][2]))
```

Comment:
Does this function need to accept a `float` input for `observation_time`? I try to avoid repetition like is done here where there are a handful of hard coded values in 2 different locations. One possible fix I thought of would be to force `observation_time` to be an ndarray even if it only has one value. I'll leave this up to you to decide.

Reply:
At the time, when I changed this, I couldn't find a way to pass in a singular float for observation time without breaking it out into this `if not` logic. But my brain was helpful today and I was able to reduce the repetition by simply casting the single input value to a list.